cement/ext/ext_smtp.py

```diff
@@ -143,91 +173,246 @@ def send(self, body: Union[str, Tuple[str, str]], **kw: Any) -> bool:
+                #       'path/simple.ext',
+                #       ('attname.ext', 'path/filename.ext'),
+                #       ('attname.ext', 'path/filename.ext', 'cidname'),
+                #       {'name': 'attname', 'path': 'path/filename.ext', cid: 'cidname'},
+                #      ]
                 if isinstance(in_path, tuple):
-                    if in_path[0] == in_path[1]:
-                        # protect against the full path being passed in
-                        alt_name = os.path.basename(in_path[0])
-                    else:
-                        alt_name = in_path[0]
+                    attname = in_path[0]
                     path = in_path[1]
+                    cid = in_path[2] if len(in_path) >= 3 else None
+                elif isinstance(in_path, dict):
+                    attname = in_path.get('name', None)
+                    path = in_path.get('path')
+                    cid = in_path.get('cid', None)
                 else:
-                    alt_name = os.path.basename(in_path)
+                    attname = None
                     path = in_path
+                    cid = None
 
                 path = fs.abspath(path)
+                if not attname:
+                    attname = os.path.basename(path)
 
-                # add attachment
+                # add attachment payload from file
                 with open(path, 'rb') as file:
-                    part.set_payload(file.read())
+                    # check for embedded image or regular attachments
+                    if cid:
+                        part = MIMEImage(file.read())
+                    else:
+                        part = MIMEBase('application', 'octet-stream')
+                        part.set_payload(file.read())
 
-                # encode and name
+                # encoder
                 encoders.encode_base64(part)
-                part.add_header(
-                    'Content-Disposition',
-                    f'attachment; filename={alt_name}',
-                )
-                msg.attach(part)
-
-        server.send_message(msg)
 
-        # FIXME: how to check success? docs don't say return type
-        # - `[ext.scrub]` [Issue #724](https://github.com/datafolklabs/cement/issues/724)
-        return True
+                # embedded inline or attachment
+                if cid:
+                    part.add_header(
+                        'Content-Disposition',
+                        f'inline; filename={attname}',
+                    )
+                    part.add_header('Content-ID', f'<{cid}>')
+                    rel.attach(part)
+                else:
+                    # attname header
+                    part.add_header(
+                        'Content-Disposition',
+                        f'attachment; filename={attname}',
+                    )
+                    msg.attach(part)
+
+        return msg
+
+    def send_by_template(self, template, data={}, **kw):
```

Comment:
This is from tokeo.ext.smtp while I enable the usage of templates to create mails.

templ.title.jinja2
templ.plain.jinja2
templ.html.jinja2

This is very usefull but I expect that this might not per default should be in Cement default extension, while you also will control the template engine with that (jinja2)

For me it can/should be out